samples/04.api/n.save-restore-scroll-position/README.md

```diff
@@ -0,0 +1,375 @@
+# Sample - Save and restore scroll position
+
+This sample shows how to save and restore scroll positions using [Web Chat hooks API](https://github.com/microsoft/BotFramework-WebChat/blob/master/docs/HOOKS.md) in composition mode.
+
+# Test out the hosted sample
+
+![Demo](https://raw.githubusercontent.com/microsoft/BotFramework-WebChat/master/samples/04.api/n.save-restore-scroll-position/demo.gif)
+
+-  [Try out MockBot](https://microsoft.github.io/BotFramework-WebChat/04.api/n.save-restore-scroll-position)
+
+# Things to try out
+
+1. Type "help" in the send box
+1. Scroll up to the midway
+1. Click "Save current position"
+1. Scroll to the top
+1. Click "Scroll to saved position"
+
+# Code
+
+> Jump to [completed code](#completed-code) to see the end-result `index.html`.
+
+This sample is based on the [04.api/m.enable-composition-mode](https://github.com/microsoft/BotFramework-WebChat/tree/master/samples/04.api/m.enable-composition-mode) sample.
+
+## Creating an overlay container
+
+An overlay container will be added to house buttons that would save/restore scroll position.
+
+### Add CSS style sheet
+
+We will put the overlay container on the top of the page.
+
+<!-- prettier-ignore-start -->
+```css
+.button-overlay {
+  background-color: White;
+  border-color: #393;
+  border-radius: 4px;
+  border-style: solid;
+  border-width: 2px;
+  box-shadow: 0 0 10px rgba(0, 0, 0, 0.05);
+  display: flex;
+  flex-direction: column;
+  left: 10px;
+  padding: 5px;
+  position: absolute;
+  top: 10px;
+}
+
+.button-overlay__button {
+  margin: 5px;
+}
+
+.button-overlay__saved-position {
+  margin: 5px;
+  padding: 0;
+}
+```
+<!-- prettier-ignore-end -->
+
+### Create a new React component
+
+Then, we will add the React component, which will house the buttons.
+
+<!-- prettier-ignore-start -->
+```js
+const ButtonOverlay = () => {
+  return (
+    <div className="button-overlay">
+    </div>
+  );
+};
+```
+<!-- prettier-ignore-end -->
+
+### Add the React component to API container
+
+Lastly, we will render the `<ButtonOverlay>` in the API container.
+
+```diff
+  ReactDOM.render(
+    <Composer directLine={window.WebChat.createDirectLine({ token })}>
+      <BasicWebChat />
++     <ButtonOverlay />
+      <SendHelpOnConnect />
+    </Composer>,
+    document.getElementById('webchat')
+  );
+```
+
+## Observing change of scroll positions
+
+The [`useObserveScrollPosition`](https://github.com/microsoft/BotFramework-WebChat/blob/master/docs/HOOKS.md#useobservescrollposition) hook subscribe to changes of scroll positions. Since scroll positions change rapidly during scrolling, the callback function should be as lightweight as possible.
```

Comment:
```suggestion
The [`useObserveScrollPosition`](https://github.com/microsoft/BotFramework-WebChat/blob/master/docs/HOOKS.md#useobservescrollposition) hook subscribes to scroll position changes. Since the scroll positions change rapidly during scrolling, the callback function should be as lightweight as possible.
```